### DISCUSSION TITLE
Flower Default User + Ingress

### DISCUSSION THREAD

**Discussion by Muchembi** [airflow-helm/charts#865] [Questions & Answers] [answered]
1. I noticed that when securing Flower, the chart only offers the option for setting the password but no user. Is there a default user which takes on the password? Or is there a way to configure both the user and password?

2. It would be great if there was an option to enable the web and flower ingresses individually. i.e.
`ingress.web.enabled` and `ingress.flower.enabled `as oppossed to the global `ingress.enabled`.

**Comment by thesuperzapper** [comment, marked as answer]
@Muchembi

1. It looks like Airflow still does not support any authentication methods for flower other than basic auth:
     - https://airflow.apache.org/docs/apache-airflow/2.9.1/security/flower.html
     - https://airflow.apache.org/docs/apache-airflow-providers-celery/stable/configurations-ref.html#celery
     - [Basic auth](https://en.wikipedia.org/wiki/Basic_access_authentication) is just passing an `Authorization` header with base64 encoded `username:password` value, so its more like a "token" than a username/password.
     - I would be interested to know if you could setup one of the [OAuth login types](https://flower.readthedocs.io/en/1.0/auth.html) in Airflow's flower, but I am not sure you can set the require start args like `--auth_provider`
2. You can raise a PR if you like, we can discuss it:
     - I think most people never notice because they run with flower disabled (as its a bit of a security risk), so the flower ingress is not created. Or they use KubernetesExecutor and so celery is not used at all.
     - But we could add a per-ingress enabled value which overrides the overall enabled value.
